### DISCUSSION TITLE
Next 11 now comes with ESLint, but doesn't appear to come bundled with @typescript-eslint/eslint-plugin

### DISCUSSION THREAD

**Discussion by AshConnolly** [vercel/next.js#27399] [Ideas]
https://nextjs.org/docs/basic-features/eslint

In Next 11 inbuilt ESLint was introduced. What is unclear is if I need to add Typescript ESLint tools like `@typescript-eslint/eslint-plugin` and `@typescript-eslint/parser`. The page above doesn't mention TypeScript and the `/node_modules/next/package.json` file has no mention of these packages. It only has "typescript" and "@babel/preset-typescript".

So presumably I do need to add these. Wouldn't it make more sense if this was baked in to help further support TypeScript? 🤔

Apologies if I've misunderstood! 
Thanks! 👍

**Comment by housseindjirdeh** [comment]
`@typescript-eslint/parser` is already [included](https://github.com/vercel/next.js/blob/canary/packages/eslint-config-next/index.js#L41-L53) in `eslint-config-next` as an override for all `.tsx` files so you don't have to add this yourself.

I opted not to include `@typescript-eslint/eslint-plugin` and leave that up to the developer to decide to add (along with `eslint:recommended` if they choose to include it as well). Do you think it would be worthwhile to include `plugin:@typescript-eslint/recommended` by default?

> What is unclear is if I need to add Typescript ESLint tools 

I have a few open PRs that are updating the documentation but I would be happy to include a better description of what TypeScript ESLint tools are included by default in the docs. Feel free to open an issue for this and you can also go ahead and submit a PR immediately after if you like :)
